code/game/machinery/outpost_electrolyzer.dm

```diff
@@ -0,0 +1,147 @@
+//allows production of hydrogen from ice chunks
+#define MOLS_PER_ICE 50
+
+/obj/machinery/mineral/electrolyzer_unloader
+	name = "ice unloading machine"
+	icon = 'icons/obj/machines/mining_machines.dmi'
+	icon_state = "unloader"
+	density = TRUE
+	input_dir = WEST
+	output_dir = EAST
+	needs_item_input = TRUE
+	processing_flags = START_PROCESSING_MANUALLY
+
+/obj/machinery/mineral/electrolyzer_unloader/pickup_item(datum/source, atom/movable/target, atom/oldLoc)
+	if(istype(target, /obj/structure/ore_box))
+		var/obj/structure/ore_box/box = target
+		for(var/obj/item/stack/ore/ice/chunk in box)
+			unload_mineral(chunk)
+	else if(istype(target, /obj/item/stack/ore/ice))
+		var/obj/item/stack/ore/chunk = target
+		unload_mineral(chunk)
+
+/obj/machinery/mineral/electrolyzer
+	name = "ice crusher"
+	desc = "Breaks down ice into hydrogen and oxygen."
+	icon = 'icons/obj/recycling.dmi'
+	icon_state = "grinder-o1"
+	input_dir = WEST
+	density = TRUE
+	use_power = IDLE_POWER_USE
+	idle_power_usage = 50
+	active_power_usage = 1000
+	max_integrity = 500
+	var/crush_damage = 1000
+	var/datum/weakref/attached_output
+
+/obj/machinery/mineral/electrolyzer/Initialize()
+	. = ..()
+	update_appearance()
+	var/static/list/loc_connections = list(
+		COMSIG_ATOM_ENTERED = PROC_REF(on_entered),
+	)
+	AddElement(/datum/element/connect_loc, loc_connections)
+
+/obj/machinery/mineral/electrolyzer/CanAllowThrough(atom/movable/mover, border_dir)
+	. = ..()
+	if(!anchored)
+		return
+	if(border_dir == input_dir)
+		return TRUE
+
+/obj/machinery/mineral/electrolyzer/proc/on_entered(datum/source, atom/movable/AM)
+	SIGNAL_HANDLER
+	set_electrolyzer_output()
+	INVOKE_ASYNC(src, PROC_REF(electrolyze), AM)
+
+
+/obj/machinery/mineral/electrolyzer/proc/electrolyze(atom/movable/electrolyze_target, sound=TRUE)
+	if(machine_stat & (BROKEN|NOPOWER))
+		return
+	if(!isturf(electrolyze_target.loc))
+		return
+	if(istype(electrolyze_target, /obj/effect))
+		return	//effects are not touchable
+
+	var/list/to_electrolyze = electrolyze_target.GetAllContents()
+
+	var/list/electrolyze = list()
+
+	var/not_electrolyzed //used to play sounds if something that isnt ice gets chucked in
+
+	var/obj/machinery/atmospherics/components/binary/electrolyzer_out/resolved_output = attached_output?.resolve()
+
+	if(!resolved_output)
+		return
+	for(var/atom/movable/target as anything in to_electrolyze)
+		if(istype(target, /obj/item/stack/ore/ice))
+			electrolyze += target
+			continue
+		not_electrolyzed = TRUE
+		if(isliving(target))
+			crush_living(target)
+	if(not_electrolyzed)
+		playsound(src, 'sound/machines/buzz-sigh.ogg', 50, FALSE, 1) // Dont play a ton of sounds for a ton of items
+	for(var/electro in electrolyze)
+		resolved_output.electrolyze_item(electro)
```

Comment:
I don't think this ever checks the item (`to_electrolyze`) itself, so you can't just throw ice in for example

you can either just add the thing itself to the list to loop through, or just have the item dump out all of its contents (thus causing them to `Enter` the electrolyzer as well).